flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -774,6 +855,43 @@ def _iter_nodes(
   for key, value in node_dict.items():
     yield from _iter_nodes(value, visited, (*path_parts, key))
 
+@tp.overload
+def compose_mapping(
+  map_ab: tp.Mapping[A, B],
+  map_bc: tp.Mapping[B, C],
+  /,
+  *,
+  invert: tp.Literal[False] = False,
+) -> dict[A, C]:
+  ...
+
+
+@tp.overload
+def compose_mapping(
+  map_ab: tp.Mapping[A, B],
+  map_bc: tp.Mapping[B, C],
+  /,
+  *,
+  invert: tp.Literal[True],
+) -> dict[C, A]:
+  ...
+
+
+def compose_mapping(
+  map_ab: tp.Mapping[A, B], map_bc: tp.Mapping[B, C], /, *, invert: bool = False
+) -> dict[A, C] | dict[C, A]:
+  if invert:
+    return {map_bc[b]: a for a, b in map_ab.items() if b in map_bc}
+  else:
+    return {a: map_bc[b] for a, b in map_ab.items() if b in map_bc}
+
+@dataclasses.dataclass(frozen=True)
+class Static(tp.Generic[A]):
```

Comment:
Please add a docstring explaining the purpose of this dataclass.

Reply:
Also worth mentioning any restrictions on `value`, if you have them, e.g. immutable, hashable etc.